src/executor/events.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * This interface describes events emitted by `TaskExecutor` through `TaskExecutor.events` object.
+ */
+export interface TaskExecutorEventsDict {
+  /**
+   * Fires when task executor is initialized and ready to be used.
+   */
+  ready: () => void;
+
+  /**
+   * Fires when task executor is about to shut down, immediately after TaskExecutor.shutdown() is called.
+   *
+   */
+  beforeend: () => void;
+
+  /**
+   * Fires when task executor is completely terminated.
```

Comment:
From the documenting perspective, you could add just a bit more information like: explaining that the pending invoices are checked and paid before this happens etc. Instead of documenting this in the examples and docs, it would be great to have this documented in the API reference.